### PR TITLE
Add components for power and ground

### PR DESCRIPTION
**Pull request by ausbin**
Using constants has always bothered me a little bit for building circuits with transistors because it feels wrong, so add components for the power and ground symbols in Patt & Patel. They look like this:

image

The same circuit looks like this in Patt & Patel 3rd ed:

image

I need to check with @pulkitgupta1217, @tmconte, and @sameer-s (and of course Mr. @ra4king) for thoughts on this. This is just my opinion and done for fun, curious what y'all think

### CODE REVIEW

**Comment by pulkitgupta1217** [comment]
I definitely agree there should be explicit elements for vdd and ground, whether they are unique components or just cosmetic changes to existing constants doesn't really matter to me. Only suggestion I have is to use the traditional circuit symbol for ground like the picture below

Power-symbols

**Comment by sameer-s** [comment]
I feel that in 2110 a lot of students are more comfortable / used to thinking in 1s and 0s rather than in more electronics terms so I'm not sure how many of them would actually use these components. However, it could still be helpful at least because a lot of students get really confused with constant pins vs. input pins and so it might be nice to have a more visually distinct option for examples and such when appropriate.

That being said, I definitely think they should be new components if added, mostly because it would be very confusing to (for example) have a "ground" component connected as the carry-in for an adder, although it would be technically correct.

**Comment by ausbin** [comment]
> I definitely agree there should be explicit elements for vdd and ground, whether they are unique components or just cosmetic changes to existing constants doesn't really matter to me. Only suggestion I have is to use the traditional circuit symbol for ground like the picture below
> 
> Power-symbols

For what it's worth, I sent the following picture to Dr. Conte and asked which he liked best:

three-transistors-walk-into-a-bar

along with the corresponding textbook diagram:

MicrosoftTeams-image (3)

He replied:
> I draw it like the textbook and like that way better than the 1, 0 thing on the left

I emailed Caleb but he hasn't replied yet, understandably

File: src/main/java/com/ra4king/circuitsim/gui/peers/wiring/PowerPeer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		super(x, y, 0, 0);
+
+		Properties properties = new Properties();
+		properties.ensureProperty(Properties.LABEL);
+		properties.ensureProperty(Properties.LABEL_LOCATION);
+		properties.mergeIfExists(props);
+
+		Power power = new Power(properties.getValue(Properties.LABEL));
+
+		setWidth(2);
```
Comment (ra4king) [inline src/main/java/com/ra4king/circuitsim/gui/peers/wiring/PowerPeer.java:39]: You can set the width and the height in the super() call.

File: src/main/java/com/ra4king/circuitsim/gui/peers/wiring/GroundPeer.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		super(x, y, 0, 0);
+
+		Properties properties = new Properties();
+		properties.ensureProperty(Properties.LABEL);
+		properties.ensureProperty(Properties.LABEL_LOCATION);
+		properties.mergeIfExists(props);
+
+		Ground ground = new Ground(properties.getValue(Properties.LABEL));
+
+		setWidth(2);
```
Comment (ra4king) [inline src/main/java/com/ra4king/circuitsim/gui/peers/wiring/GroundPeer.java:39]: You can set the width and the height in the super() call.

File: src/main/java/com/ra4king/circuitsim/gui/peers/wiring/GroundPeer.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import javafx.scene.image.Image;
+import javafx.util.Pair;
+
+/**
+ * @author Austin Adams and Roi Atalla
+ */
+public class GroundPeer extends ComponentPeer<Ground> {
+	public static void installComponent(ComponentManagerInterface manager) {
+		manager.addComponent(new Pair<>("Wiring", "Ground"),
+		                     new Image(ConstantPeer.class.getResourceAsStream("/images/Ground.png")),
```
Comment (ra4king) [inline src/main/java/com/ra4king/circuitsim/gui/peers/wiring/GroundPeer.java:25]: Use GroundPeer here

File: src/main/java/com/ra4king/circuitsim/gui/peers/wiring/PowerPeer.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import javafx.scene.image.Image;
+import javafx.util.Pair;
+
+/**
+ * @author Austin Adams and Roi Atalla
+ */
+public class PowerPeer extends ComponentPeer<Power> {
+	public static void installComponent(ComponentManagerInterface manager) {
+		manager.addComponent(new Pair<>("Wiring", "Power"),
+		                     new Image(ConstantPeer.class.getResourceAsStream("/images/Power.png")),
```
Comment (ra4king) [inline src/main/java/com/ra4king/circuitsim/gui/peers/wiring/PowerPeer.java:25]: Use PowerPeer here

**Review by ra4king — CHANGES_REQUESTED** [review]

**Comment by ausbin** [comment]
Thanks for those catches, is this better?

**Comment by ra4king** [comment]
I'm not a huge fan of the "T" and triangle shape but if this is what Conte wants...

**Review by ra4king — APPROVED** [review]

**Comment by ausbin** [comment]
> I'm not a huge fan of the "T" and triangle shape but if this is what Conte wants...

@ra4king I don't think you have the same obligation to obey Patt & Patel that he does, if that makes sense. In my opinion, the appearance of the component could differ between vanilla CircuitSim and 2110 CircuitSim, even if the size & port locations don't. (If we want to go that route, I can make the PRs in the right places to avoid tormenting the TAs)

For ground, do you like this symbol more? (I have that written up already)

image

And for power, were you thinking something like this?

image

This would match what Logisim does: http://www.cburch.com/logisim/docs/2.7/en/html/libs/wiring/const01.html

**Comment by ra4king** [comment]
No it's ok, I googled a bit on this subject and the current ground symbol is fine. Let's add "VDD" to the power symbol?

**Pull request merged**